UPGRADING.md

```diff
@@ -0,0 +1,184 @@
+# Upgrading Notes (2.x to 3.x)
+
+This document captures notable changes from v2 SDK to v3 SDK, as known as modular JavaScript SDK for AWS.
+
+Because v3 is a modular rewrite of v2, some basic conceptions are different between v2 and v3. You can learn about
+these changes in our fast growing list of [blog posts](https://aws.amazon.com/blogs/developer/category/developer-tools/aws-sdk-for-javascript-in-node-js/). Among the blog posts, The following 2 will get you up-to-speed:
+
+- [modular packages](https://aws.amazon.com/blogs/developer/modular-packages-in-aws-sdk-for-javascript/)
+- [middleware stack](https://aws.amazon.com/blogs/developer/first-class-typescript-support-in-modular-aws-sdk-for-javascript/)
+
+Bellow are a summary of interface changes from v2 to v3 SDK. The goal is to help you easily find the v3 equivalents of
+the v2 APIs you are already familiar with.
+
+## Client Constructors
+
+This list is indexed by [parameters in v2 SDK](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html). For
+options shown as `Planned` in v3, they are confirmed to be supported, but no timeline can be shared at the moment. They
+might not have the same name either.
+
+- [`computeChecksums`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#computeChecksums-property)
+  - **v2**: Whether to compute MD5 checksums for payload bodies when the service accepts it (currently supported in S3
+    only).
+  - **v3**: Not available. Planned.
+- [`convertResponseTypes`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#convertResponseTypes-property)
+  - **v2**: Whether types are converted when parsing response data.
+  - **v3**: **Deprecated**. This option is considered not type-safe because it doesn't convert the types like time stamp
+    or base64 binaries from the JSON response.
+- [`correctClockSkew`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#correctClockSkew-property)
+  - **v2**: Whether to apply a clock skew correction and retry requests that fail because of an skewed client
+    clock.
+  - **v3**: **Deprecated**. SDK _always_ applies a clock skew correction.
+- [`systemClockOffset`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#systemClockOffset-property)
+  - **v2**: An offset value in milliseconds to apply to all signing times.
+  - **v3**: No change.
+- [`credentials`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#credentials-property)
+  - **v2**: The AWS credentials to sign requests with.
+  - **v3**: No change. It can also be an async function that returns a credential.
```

Comment:
Singular for credentials is credentials, as it has multiple values like accessKeyId, secretAccessKey etc.
```suggestion
  - **v3**: No change. It can also be an async function that returns credentials.
```